test/performance/README.md

```diff
@@ -111,13 +111,86 @@ To generate the sample workload for KubeStellar performance experiments proceed
     - 10 config maps containing a 2048-character random string
 
 
-
-
 5. Clean up the generated workload Kubernetes API objects from your cluster:
 
    ```bash
    $ cd $CL2_DIR
    $ ./cleanup.sh
    ```
 
-### Collect the creation & update timestamps for workload benchmark (OPTIONAL) (TO DO: WORK-IN-PROGRESS)
\ No newline at end of file
+
+### Collect the creation & update timestamps for workload benchmark (OPTIONAL)
+
+
+To collect the creation and status_update timestamps for the benchmark workload objects proceed as following:  
+
+1. Run the metrics collection script:
+
+   a) cd into the `test/performance` directory from your local copy of the KubeStellar repo, for example:
+
+   ```bash
+   $ cd .$HOME/kubestellar/test/performance
+   ```
+
+   b) Run the metrics collection script:
+
+   ```bash 
+   $ python3 metrics_collector.py <kubeconfig> <wds_ctx> <its_ctx> <wec_ctx> <num_ns> <output_dir> 
+   ```
+
+   For example:
+   
+   ```bash 
+   $ python3 metrics_collector.py $HOME/.kube/config wds1 its1 cluster1 2 $HOME/data
+   ```
+
+   Below is a detailed explanation of the input parameters:
+   - `kubeconfig`: path to the kubeconfig file, e.g., *$HOME/.kube/config*
+   - `wds_ctx`: name of the context for the target WDS, e.g., *"wds1"*
+   - `its_ctx`: name of the context for the target ITS, e.g., *"its1"*
+   - `wec_ctx`: name of the context for the target WEC, e.g., *"cluster1"*
+   - `num_ns`: number of namespaces created in your experiment, e.g., *2*
+   - `output_dir`: path to the directory for the output data files, e.g., *$HOME/data/*
+
+   For each namespace, the creation and status_update timestamps will be collected for the workload objects created in `wds1`, manifestwork & workstatus objects created in `its1`, and workload & appliedmanifestworks objects created in `wec1`, for example: let's assume that `num_ns=2`, then the following directories and files will be created: 
+
+   ```bash 
+      $tree
+      .
+      ├── perf-exp-0
+      │   ├── appliedmanifestworks
+      │   ├── configmaps-wds1
+      │   ├── configmaps-wec
+      │   ├── deployments-wds1
+      │   ├── deployments-wec
+      │   ├── manifestworks
+      │   ├── secrets-wds1
+      │   ├── secrets-wec
+      │   ├── services-wds1
+      │   ├── services-wec
+      │   └── workstatuses
+      └── perf-exp-1
+         ├── appliedmanifestworks
+         ├── configmaps-wds1
+         ├── configmaps-wec
+         ├── deployments-wds1
+         ├── deployments-wec
+         ├── manifestworks
+         ├── secrets-wds1
+         ├── secrets-wec
+         ├── services-wds1
+         ├── services-wec
+         └── workstatuses
+   ```
+
+   The generated output file has the following structure:
+
+   ```bash
+   <obj_name> <obj_creation_time> <obj_status_update_time> <obj_status_condition>  <obj_status_condition>  <obj_controller_manager>
```

Comment:
(A) Why is `<obj_status_condition>` listed twice on this line? It appears only once in the example below.

(B) What does "obj_controller_manager" mean? The example below is not illuminating.

Reply:
(A) It was a typo - fixed it

(B) obj_controller_manager - it's the name of the controller that updates the status for the k8s object.

Reply:
Thank you for the explanation. I think it would be good to have an explanation in the document.

Reply:
Fixed!